psl/psl-core/src/validate/validation_pipeline/validations/relation_indexes.rs

```diff
@@ -0,0 +1,78 @@
+use diagnostics::DatamodelWarning;
+use parser_database::{
+    ast::{FieldId, ModelId},
+    walkers::{RelationFieldWalker, Walker},
+};
+
+use crate::{datamodel_connector::RelationMode, validate::validation_pipeline::context::Context};
+
+// { x_1, x_2, ..., x_n } is left-wise included in { y_1, y_2, ..., y_m } if and only if
+// n <= m and x_i = y_i for all i in [1, n].
+fn is_left_wise_included<T>(item: &[T], group: &[T]) -> bool
+where
+    T: PartialEq,
+{
+    group.iter().take(item.len()).eq(item.iter())
+}
+
+pub(super) fn validate_missing_relation_indexes(
+    model: Walker<'_, ModelId>,
+    relation_field: RelationFieldWalker<'_>,
+    ctx: &mut Context<'_>,
+) {
+    let is_provider_mongodb = ctx
+        .datasource
+        .map(|datasource| datasource.active_provider == "mongodb")
+        .unwrap_or(false);
+
+    if is_provider_mongodb || ctx.relation_mode != RelationMode::Prisma {
+        return;
+    }
```

Comment:
`is_provider_mongodb` doesn't belong here, the core shouldn't know about specific connectors.

If we want to skip that validation for specific connectors, we should add a `fn should_suggest_indexes_on_referencing_fields_relation_mode_prisma(&self) -> bool` method to `Connector`, or move that validation inside the connectors.

Reply:
Ok, I would simplify that method's name to `should_suggest_missing_referencing_fields_indexes`, as the check for `relationMode = "prisma"` should be performed separately from the "connector requires referencing fields' index validation" check